docs/modeling/bounding-boxes.rst

```diff
@@ -0,0 +1,149 @@
+individual model subclasses that have a ``bounding_box_default`` function 
+defined. ``bounding_box_default`` returns the minimum rectangular region 
+symmetric about the position that fully contains the model if the model has a
+finite extent. If a model does not have a finite extent, the choice for the 
+``bounding_box_default`` limits is noted in the docstring. For example, see
+`astropy.modeling.functional_models.Gaussian2D.bounding_box_default`.
+
+The default function can also be set to any callable. This is particularly 
+useful for fitting ``custom_model`` or ``CompoundModel`` instances. 
+
+    >>> from astropy.modeling import custom_model
+    >>> def ellipsoid(x, y, z, x0=0, y0=0, z0=0, a=2, b=3, c=4, amp=1):
+    ...     rsq = ((x-x0)/a) ** 2 + ((y-y0)/b) ** 2 + ((z-z0)/c) ** 2
+    ...     val = (rsq < 1) * amp
+    ...     return val
+    ...
+    >>> def ellipsoid_bb(self):
+    ...     return ((self.z0 - self.c, self.z0 + self.c),
+    ...             (self.y0 - self.b, self.y0 + self.b),
+    ...             (self.x0 - self.a, self.x0 + self.a))
+    ...
+    >>> Ellipsoid3D = custom_model(ellipsoid)
+    >>> Ellipsoid3D.bounding_box_default = ellipsoid_bb
+    >>> model = Ellipsoid3D()
+    >>> model.bounding_box = 'auto'
+    >>> model.bounding_box
+    ((-4.0, 4.0), (-3.0, 3.0), (-2.0, 2.0))
+
+Efficient evaluation with ``render_model``
+------------------------------------------
+
+When a model is evaluated over a range much larger than the model itself, it may 
+be prudent to use `astropy.modeling.render_model` if efficiency is a concern. 
+The ``render_model`` function can be used to evaluate a model on an array of the 
+same dimensions. If no array is given, ``render_model`` will return a "postage 
+stamp" array corresponding to the bounding box region. However, if 
+``bounding_box`` is `None` an image or coordinates must be passed. 
+
+In this example, we generate a 300x400 pixel image of 100 2D 
+Gaussian sources both with and without using bounding boxes. Using bounding 
+boxes, the evaluation speed increases by approximately a factor of 10 with 
+negligible loss of information.
+
+.. plot::
+    :include-source:
+    
+	import numpy as np
+	from time import time
+	from astropy.modeling import models, render_model
+
+	import matplotlib as mpl
+	import matplotlib.pyplot as plt
+	from astropy.visualization import astropy_mpl_style
+	astropy_mpl_style['axes.grid'] = False
+	astropy_mpl_style['axes.labelcolor'] = 'k'
+	mpl.rcParams.update(astropy_mpl_style) 
+
+	np.random.seed(0)
+
+	imshape = (300, 400)
+	nsrc = 100
+
+	model_params = [
+	    dict(amplitude = np.random.uniform(0, 1),
+	         x_mean = np.random.uniform(0, imshape[1]),
+	         y_mean = np.random.uniform(0, imshape[0]),
+	         x_stddev = np.abs(np.random.uniform(3, 6)),
+	         y_stddev = np.abs(np.random.uniform(3, 6)),
+	         theta = np.random.uniform(0, 2 * np.pi))
+	    for i in range(nsrc)]
```

Comment:
This could still use `_` instead of `i` since `i` isn't actually used, but no matter (I think it's clearer since it emphasises that the loop variable is not actually used)